hpy/devel/__init__.py

```diff
@@ -201,18 +201,37 @@ def handle_hpy_ext_modules(dist, attr, hpy_ext_modules):
 _HPY_UNIVERSAL_MODULE_STUB_TEMPLATE = """
 # DO NOT EDIT THIS FILE!
 # This file is automatically generated by hpy
-
 def __bootstrap__():
-
-    from sys import modules
-    from os import environ
-    from pkg_resources import resource_filename
+    from sys import modules, version_info
+    from os import path, environ
     from hpy.universal import _load_bootstrap
-    ext_filepath = resource_filename(__name__, {ext_file!r})
-    m = _load_bootstrap({module_name!r}, __name__, __package__, ext_filepath,
-                        __loader__, __spec__, environ)
+
+    ext_name = {ext_file!r}
+    module_name = {module_name!r}
+
+    if version_info < (3, 10):
+        import importlib_resources as resources
+    else:
+        from importlib import resources
+
+    if not __package__:
+        # directly called with python -m
+        ext_filepath = path.join(path.dirname(__file__), ext_name)
+    else:
+        ext_filepath = resources.files(__package__).joinpath(ext_name)
```

Comment:
Are you sure it's okay for `ext_filepath` to be a different type `str` vs `Traversable` based on whether `__package__` is falsy or not?

Reply:
Never mind, you're stringifying it later.

Reply:
yes, see line 227